tools/moth.py

```diff
@@ -279,64 +153,52 @@ def make_answer(self, word_count, sep=' '):
         :returns: The answer string
         """
 
-        answer = sep.join(self.rand.sample(self.ANSWER_WORDS, word_count))
-        self['answer'] = answer
-
+        answer = sep.join(self.rand.sample(ANSWER_WORDS, word_count))
+        self.answers.append(answer)
         return answer
 
-    def htmlify(self):
+    def get_body(self):
+        return self.body.getvalue()
+
+    def html_body(self):
         """Format and return the markdown for the puzzle body."""
-        return mistune.markdown(self.body)
-
-    def publish(self):
-        obj = {
-            'author': self['author'],
-            'hashes': self['hashes'],
-            'body': self.htmlify(),
-        }
-        return obj
-
-    def secrets(self):
-        obj = {
-            'answers': self['answers'],
-            'summary': self['summary'],
-        }
-        return obj
-
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser(description='Build a puzzle category')
-    parser.add_argument('puzzledir', nargs='+', help='Directory of puzzle source')
-    args = parser.parse_args()
-
-    for puzzledir in args.puzzledir:
-        puzzles = {}
-        secrets = {}
-        for puzzlePath in glob.glob(os.path.join(puzzledir, "*.moth")):
-            filename = os.path.basename(puzzlePath)
-            points, ext = os.path.splitext(filename)
-            points = int(points)
-            puzzle = Puzzle(puzzlePath, "test")
-            puzzles[points] = puzzle
-
-        for points in sorted(puzzles):
-            puzzle = puzzles[points]
-            print(puzzle.secrets())
+        return mistune.markdown(self.get_body())
+
+    def hashes(self):
+        "Return a list of answer hashes"
+
+        return [djbhash(a) for a in self.answers]
 
 
 class Category:
     def __init__(self, path, seed):
         self.path = path
         self.seed = seed
         self.pointvals = []
-        for fpath in glob.glob(os.path.join(path, "[0-9]*")):
-            pn = os.path.basename(fpath)
-            points = int(pn)
-            self.pointvals.append(points)
+        self.catmod = None
+
+        if os.path.exists(os.path.join(path, 'category.py')):
+            self.catmod = importlib.machinery.SourceFileLoader(
+                'catmod',
+                os.path.join(path, 'category.py')).load_module()
+            self.pointvals = self.catmod.points[:]
```

Comment:
`self.pointvals.extend(self.catmod.points)` is more pythonic. using `self.catmod.points[:]` provides only a shallow copy. If `self.catmod.points`  has something evil in it, it will explode immedately with `list.extend()` whereas the shallow copy of the list-interface-implementor will explode later, if it's going to explode.